src/agent/coverage/tests/snapshot.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[test]
+#[cfg(target_os = "windows")]
+fn windows_snapshot_tests() {
+    use coverage::{binary::DebugInfoCache, source::Line, AllowList};
+    use debuggable_module::path::FilePath;
+    use std::fmt::Write;
+    use std::time::Duration;
+
+    insta::glob!("windows", "*.cpp", |path| {
+        let file_name = path.file_name().unwrap();
+
+        // locate appropriate compiler
+        let mut cl_exe =
+            cc::windows_registry::find("x86_64-pc-windows-msvc", "cl.exe").expect("finding cl.exe");
+
+        // path will have \\?\ prefix, cl.exe doesn't like it
+        let input_path = dunce::canonicalize(path).unwrap();
+
+        // directory to compile into
+        let build_in = tempfile::tempdir().expect("creating tempdir");
+
+        let output = cl_exe
+            .arg("/EHsc")
+            .arg("/Zi")
+            .arg("/O2")
+            .arg(&input_path)
+            .current_dir(build_in.path())
+            .spawn()
+            .expect("launching compiler")
+            .wait_with_output()
+            .expect("waiting for compiler to finish");
+
+        assert!(output.status.success(), "cl.exe failed: {:?}", output);
+
+        let exe_name = {
+            let mut cwd = build_in.path().to_path_buf();
+            cwd.push(file_name);
+            cwd.set_extension("exe");
+            cwd
+        };
+
+        // filter to just the input test file:
+        let source_allowlist = AllowList::parse(&input_path.to_string_lossy()).unwrap();
+
+        let exe_cmd = std::process::Command::new(&exe_name);
+        let recorded = coverage::CoverageRecorder::new(exe_cmd)
+            .timeout(Duration::from_secs(120))
+            .debuginfo_cache(DebugInfoCache::new(source_allowlist.clone()))
```

Comment:
Consider pre-seeding the cache with its own (maybe manual!) time measurement, and failing the test if it takes too long. Maybe for future work / a separate test?

Reply:
This can actually take a loong time if you pull in the C++ STL without filtering symbols 😄